docs/decisions/0017-reimplement-asset-processing.rst

```diff
@@ -222,6 +225,54 @@ The three top-level edx-platform asset processing actions are *build*, *collect*
 
        Note: This adds a Python dependency to build-assets.sh. However, we could be clear that watchman is an *optional* dependency of build-assets.sh which enables the optional ``--watch`` feature. This would keep the *build* action Python-free. Alternatively, watchman is also available Python-free via apt and homebrew.
 
+Build Configuration
+-------------------
+
+``scripts/build-assets.sh`` will accept various command-line options to configure the build. It will also accept the same options in the form of the ``OPENEDX_BUILD_ASSETS_OPTS`` enviroment variable. Options from the environment variable will be processed first, and then overridden by options provided on the command line. The environment variable allows distributions like Tutor to seed the build script with "defaults" in the event that the upstream defaults are not sufficient, while still allowing individual operators to run the script with whichever options they like.
+
+As a concrete example, the default value of ``--theme-dirs`` will be ``''`` (that is: no themes) and the default value of ``--static-root`` will be ``./test_root/static``. Neither of those are suitable for Tutor. Instead, Tutor will set the environment variable in its Dockerfile::
+
+  ...
+  ENV OPENEDX_BUILD_ASSETS_OPTS '--theme-dirs /openedx/themes --static-root /openedx/staticfiles'
+  ...
+
+Later, in the container, a user might run::
+
+  # This would search for themes in /openedx/themes
+  # and use /openedx/staticfiles as the static root:
+  scripts/build-assets.sh
+
+  # This would search for themes in ./mythemes
+  # but use /openedx/staticfiles as the static root:
+  scripts/build-assets.sh --theme-dirs ./mythemes
+
+Furthermore, to facilitate a Python-free build reimplementation, we will remove two Django settings related to assets. These settings have never worked in Tutor, and 2U states that edx.org does not use them. However, on the off chance that some community operators rely on them, there exist alternative configuration methods for each, which will work both with and without Tutor:
```

Comment:
> However, on the off chance that some community operators rely on them

Whoops, I should have updated this, since Jeremy pointed out that OpenCraft probably uses JS_ENV_EXTRA_CONFIG. Anyway, I reached out in `#opencraft` to make sure they see this.